docs/semgrepignore-v2-reference.md

```diff
@@ -0,0 +1,155 @@
+---
+slug: semgrepignore-v2-reference
+append_help_link: true
+description: |
+  Reference of the semgrepignore file fitering mechanism for Semgrep
+hide_title: true
+title: Semgrepignore reference (v2)
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Semgrepignore reference (v2)
+
+This document covers the Semgrepignore target filtering system that is
```

Comment:
Do we need to capitalize semgrepignore?

Reply:
To clarify: `gitignore` as a feature, not just the filename, **isn't** capitalized, so I'd prefer we follow their convention and also keep `semgrepignore` small caps.

Reply:
> Do we need to capitalize semgrepignore?

I think it's more logical. Writing "`gitignore`" is confusing imo.

My idea here is to capitalize "Gitignore" and "Semgrepignore" when referring to the whole feature as an ordinary proper noun (it's a proper noun because it was made up and refers to a unique thing). This implies using an ordinary font, not a monospace font. On the other hand, any text that's entered into the computer or read from it is quoted verbatim using a monospace font to indicate this. I think it's a common assumption. This applies to command names, full commands, file paths, and any text copy-pasted from a console. File names such as `.gitignore` or `.semgrepignore` as written as such (monospace font). What doesn't make sense to me is that the [Gitignore help page](https://git-scm.com/docs/gitignore) writes it as "`gitignore`" (lowercase, monospace font) even though there's no such thing as a `gitignore` command or file.